### PR TITLE
Added User-Agent interceptor and properties file

### PR DESCRIPTION
**Pull request by ricellis**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant-labs/kafka-connect-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.md](https://github.com/cloudant-labs/kafka-connect-cloudant/blob/master/CHANGES.md)
- [x] You have completed the PR template below:

## What

Added the `UserAgentInterceptor` to the client build.

## How

Consolidated methods that build CloudantClient so UA interceptor can be added in a single place.
Added `UserAgentInterceptor` initialization based on properties file.
Added gradle for property file generation for inclusion in jar.

## Testing

No new automated tests, but validated the User-Agent header was as expected.

## Issues

N/A

### CODE REVIEW

File: src/main/java/com/ibm/cloudant/kafka/common/utils/JavaCloudantUtil.java [hunk 1]
```diff
@@ -123,35 +126,46 @@ public static JSONArray batchWrite(String url, String userName, String password,
...
+	public static CloudantClient getClientInstanceFromDBUrl(String url, String username, String
+			password) {
+		CloudantClient cloudantClient = null;
 		// Create a new CloudantClient instance for account endpoint account.cloudant.com
 		// url: https://account.cloudant.com/dbname
 		// length of "://" is 3
 		if (url != null) {
-			String urlWithoutProtocal = url.substring(url.indexOf("://") +3);
-			
+			String urlWithoutProtocal = url.substring(url.indexOf("://") + 3);
```
Comment (emlaver) [inline src/main/java/com/ibm/cloudant/kafka/common/utils/JavaCloudantUtil.java]: Can we fix protocal to `urlWithoutProtocol`?

Comment (emlaver) [reply src/main/java/com/ibm/cloudant/kafka/common/utils/JavaCloudantUtil.java]: Should we consider using `new URI(string)` to get the account info and replace `indedOf() + 3`?

File: src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java [hunk 2]
```diff
@@ -201,20 +200,7 @@ public void start(Map<String, String> props) {
...
-                    .username(userName)
-                    .password(password)
-                    .build();
-
-            // Create a database instance
-            String cantDBName = url.substring(url.lastIndexOf("/") + 1);
-
-            // Create a database instance
-            cantDB = cantClient.database(cantDBName, false);
+            cantDB = JavaCloudantUtil.getDBInst(url, userName, password, false);
```
Comment (emlaver) [inline src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java]: The variable name `cantDB` should be updated to something more descriptive

File: src/main/java/com/ibm/cloudant/kafka/common/utils/JavaCloudantUtil.java (hunk 1 above)
Comment (ricellis) [reply src/main/java/com/ibm/cloudant/kafka/common/utils/JavaCloudantUtil.java]: Fixed in 6107899e20e886a0351496fb6d40b444823a9aaf

File: src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java (hunk 2 above)
Comment (ricellis) [reply src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java]: Fixed in a35d0bd0e3ef803441c52b25163f1903b9deab90

**Pull request merged**